openclimatefix/pv-site-datamodel
# Add forecast timestamp column

_I recommend reviewing commit by commit_

* Improve our alembic setup
  * Add alembic as a dependency
  * Move all alembic stuff to `sdk/python/alembic` (this is the typical setup)
  * Remove our custom `app.py` to manage the migrations and use alembic's CLI instead
  * Update our alembic `README.md` accordingly and also add some meat around the different steps, after going through them for a migration
* Test the migrations
 
* Add a `timestamp_utc` column in the `forecasts` table. This is the time at which we are making the prediction, *according to the model*. This is not necessarily the actual time at which we run the code, for instance if we wanted to backfill forecasts, this date could be in the past.
* Remove the redundant `site_uuid` column in `forecast_values` (it is already in `forecast` and we need to `JOIN` on that table anyway for `timestamp_utc`. I think having the redundant value was a bad idea after all.
* Add a test to make sure we can do the typical query of getting the latest forecasts for a list of sites. This can also serve as an example.

# TODO
- [x] Adjust the migration to use the `created_utc` date, otherwise the migration will crash because the column is set to `NOT NULL`.
- [x] Add test for read function and fix it - this PR breaks it as it is
